front/lib/api/chat.ts

```diff
@@ -108,13 +110,15 @@ export async function getChatSession(
     created: chatSession.createdAt.getTime(),
     sId: chatSession.sId,
     title: chatSession.title,
+    visibility: chatSession.visibility,
   };
 }
 
 export async function upsertChatSession(
   auth: Authenticator,
   sId: string,
-  title: string | null
+  title: string | null,
+  visibility: ChatSessionVisibility | null
```

Comment:
Feel free to not address here, but I think it's starting to make sense to refactor this one to take a single object argument (as `upsertChatSession(auth, sId, null, null)` is not super readable)